### ISSUE TITLE
orderBy doesn't accept a subquery

### ISSUE THREAD

**Issue by koskimas** [knex/knex#3489]
# Environment

Knex version: all
Database + version: all
OS: all

# Feature discussion / request

Currently this doesn't work:

```js
knex('table').orderBy(knex('otherTable').whereColumn('table.id', 'otherTable.someId').select('blah'))
```

This does work correctly

```js
knex('table').orderByRaw('?', knex('otherTable').whereColumn('table.id', 'otherTable.someId').select('blah'))
```

knex doesn't seem to add parentheses around the subquery in the first case.

Here's a reproduction:

```js
const Knex = require('knex');

async function main() {
  await createSchema();
  await knex('persons')
    .debug()
    .orderBy(
      knex('persons as p')
        .whereColumn('persons.id', 'p.id')
        .select('p.id')
    );
}

///////////////////////////////////////////////////////////////
// Database
///////////////////////////////////////////////////////////////

const knex = Knex({
  client: 'sqlite3',
  useNullAsDefault: true,
  debug: false,
  connection: {
    filename: ':memory:'
  }
});

///////////////////////////////////////////////////////////////
// Schema
///////////////////////////////////////////////////////////////

async function createSchema() {
  await knex.schema.dropTableIfExists('persons');
  await knex.schema.createTable('persons', table => {
    table.increments('id').primary();
  });
}

main()
  .then(() => {
    console.log('success');
    return knex.destroy();
  })
  .catch(err => {
    console.error(err);
    return knex.destroy();
  });
```

**Comment by kibertoad** [comment]
Released in 0.20.0

**Comment by koskimas** [comment]
Awesome! Thank you @HurSungYun and @kibertoad
